### 공공 데이터 분석 (with folium)
* 전국 전통시장 표준 데이터

#### 01. 라이브러리 불러오기

In [3]:
import pandas as pd
import folium

# 기본 encoding은 utf-8, 에러 발생시 : euc-kr로 설정
market = pd.read_csv('전국전통시장표준데이터.csv', encoding='euc-kr')
market.head()

시장명 시장유형                 소재지도로명주소                소재지지번주소 시장개설주기  \
0  해도5시장(형산시장)  상설장                      NaN  경상북도 포항시 남구 대도동 143-1     매일   
1       상대종합시장  상설장  경상북도 포항시 남구 양학천로76번길 13                    NaN     매일   
2       동해종합시장  상설장                      NaN  경상북도 포항시 남구 해도동 435-2     매일   
3         송도시장  상설장                      NaN  경상북도 포항시 남구 송도동 440-6     매일   
4         청림시장  상설장                      NaN  경상북도 포항시 남구 청림동 588-2     매일   

          위도          경도    점포수                취급품목          사용가능상품권 홈페이지주소  \
0  36.012702  129.366493   40.0  농산물 생필품, 수산물, 잡화 등  포항사랑상품권, 온누리상품권    NaN   
1  36.019933  129.361108   41.0  농산물 생필품, 수산물, 잡화 등  포항사랑상품권, 온누리상품권    NaN   
2  36.025808  129.370949  110.0  농산물 생필품, 수산물, 잡화 등  포항사랑상품권, 온누리상품권    NaN   
3  36.032826  129.375281   56.0  농산물 생필품, 수산물, 잡화 등  포항사랑상품권, 온누리상품권    NaN   
4  35.996576  129.410356   13.0  농산물 생필품, 수산물, 잡화 등  포항사랑상품권, 온누리상품권    NaN   

  공중화장실보유여부 주차장보유여부  개설연도 전화번호     데이터기준일자   제공기관코드 제공기관명  
0         Y       N  1990  NaN  2020-02-25  6470000  경상북도  
1         Y       N  1986  NaN  2020-02-25  6470000  경상북도  
2         Y       Y  1981  NaN  2020-02-25  6470000  경상북도  
3         Y       Y  1982  NaN  2020-02-25  6470000  경상북도  
4         Y       N  1985  NaN  2020-02-25  6470000  경상북도

In [4]:
### 데이터 살펴보기
print(market.columns)
print(market.shape)
print(market.info() )

Index(['시장명', '시장유형', '소재지도로명주소', '소재지지번주소', '시장개설주기', '위도', '경도', '점포수',
       '취급품목', '사용가능상품권', '홈페이지주소', '공중화장실보유여부', '주차장보유여부', '개설연도', '전화번호',
       '데이터기준일자', '제공기관코드', '제공기관명'],
      dtype='object')
(3489, 18)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3489 entries, 0 to 3488
Data columns (total 18 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   시장명        3489 non-null   object 
 1   시장유형       3489 non-null   object 
 2   소재지도로명주소   3197 non-null   object 
 3   소재지지번주소    3088 non-null   object 
 4   시장개설주기     3489 non-null   object 
 5   위도         3411 non-null   float64
 6   경도         3411 non-null   float64
 7   점포수        3406 non-null   float64
 8   취급품목       3275 non-null   object 
 9   사용가능상품권    1749 non-null   object 
 10  홈페이지주소     485 non-null    object 
 11  공중화장실보유여부  3489 non-null   object 
 12  주차장보유여부    3489 non-null   object 
 13  개설연도       1957 non-null   object 
 14  전화번호       1835 non-null   

In [5]:
market.isnull().sum()

시장명             0
시장유형            0
소재지도로명주소      292
소재지지번주소       401
시장개설주기          0
위도             78
경도             78
점포수            83
취급품목          214
사용가능상품권      1740
홈페이지주소       3004
공중화장실보유여부       0
주차장보유여부         0
개설연도         1532
전화번호         1654
데이터기준일자         0
제공기관코드          0
제공기관명           0
dtype: int64

###### 전체 2882개 데이터중에 107개 정도가 결측치
##### 전체 3489개 데이터중에 78개 정도가 결측치

In [6]:
market_name = market['시장명']
market_lati = market['위도']
market_long = market['경도']

market.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3489 entries, 0 to 3488
Data columns (total 18 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   시장명        3489 non-null   object 
 1   시장유형       3489 non-null   object 
 2   소재지도로명주소   3197 non-null   object 
 3   소재지지번주소    3088 non-null   object 
 4   시장개설주기     3489 non-null   object 
 5   위도         3411 non-null   float64
 6   경도         3411 non-null   float64
 7   점포수        3406 non-null   float64
 8   취급품목       3275 non-null   object 
 9   사용가능상품권    1749 non-null   object 
 10  홈페이지주소     485 non-null    object 
 11  공중화장실보유여부  3489 non-null   object 
 12  주차장보유여부    3489 non-null   object 
 13  개설연도       1957 non-null   object 
 14  전화번호       1835 non-null   object 
 15  데이터기준일자    3489 non-null   object 
 16  제공기관코드     3489 non-null   object 
 17  제공기관명      3489 non-null   object 
dtypes: float64(3), object(15)
memory usage: 490.8+ KB


#### 02. 데이터 살펴보기
##### 결측치를 제외한 행만 가져오기

In [7]:
market['위도'].notnull()

0       True
1       True
2       True
3       True
4       True
        ... 
3484    True
3485    True
3486    True
3487    True
3488    True
Name: 위도, Length: 3489, dtype: bool

In [8]:
market_1 = market.loc[ market['위도'].notnull() , :]
market_1.shape

(3411, 18)

In [9]:
market_1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3411 entries, 0 to 3488
Data columns (total 18 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   시장명        3411 non-null   object 
 1   시장유형       3411 non-null   object 
 2   소재지도로명주소   3136 non-null   object 
 3   소재지지번주소    3030 non-null   object 
 4   시장개설주기     3411 non-null   object 
 5   위도         3411 non-null   float64
 6   경도         3411 non-null   float64
 7   점포수        3328 non-null   float64
 8   취급품목       3197 non-null   object 
 9   사용가능상품권    1685 non-null   object 
 10  홈페이지주소     461 non-null    object 
 11  공중화장실보유여부  3411 non-null   object 
 12  주차장보유여부    3411 non-null   object 
 13  개설연도       1891 non-null   object 
 14  전화번호       1777 non-null   object 
 15  데이터기준일자    3411 non-null   object 
 16  제공기관코드     3411 non-null   object 
 17  제공기관명      3411 non-null   object 
dtypes: float64(3), object(15)
memory usage: 506.3+ KB


In [10]:
market_1.columns

Index(['시장명', '시장유형', '소재지도로명주소', '소재지지번주소', '시장개설주기', '위도', '경도', '점포수',
       '취급품목', '사용가능상품권', '홈페이지주소', '공중화장실보유여부', '주차장보유여부', '개설연도', '전화번호',
       '데이터기준일자', '제공기관코드', '제공기관명'],
      dtype='object')

In [11]:
### 필요한 정보(특징)만 가져오기
sel = ['시장명', '시장유형', '시장개설주기', '위도', '경도', 
       '사용가능상품권', '공중화장실보유여부', '주차장보유여부', '개설연도',
       '데이터기준일자', '제공기관명']

market_2 =  market_1[sel]
market_2

시장명 시장유형 시장개설주기         위도          경도          사용가능상품권  \
0     해도5시장(형산시장)  상설장     매일  36.012702  129.366493  포항사랑상품권, 온누리상품권   
1          상대종합시장  상설장     매일  36.019933  129.361108  포항사랑상품권, 온누리상품권   
2          동해종합시장  상설장     매일  36.025808  129.370949  포항사랑상품권, 온누리상품권   
3            송도시장  상설장     매일  36.032826  129.375281  포항사랑상품권, 온누리상품권   
4            청림시장  상설장     매일  35.996576  129.410356  포항사랑상품권, 온누리상품권   
...           ...  ...    ...        ...         ...              ...   
3484       수영팔도시장  상설장     매일  35.168782  129.116897              NaN   
3485     현대종합상가시장  상설장     매일  35.168408  129.120732              NaN   
3486         거성시장  상설장     매일  35.190006  129.068220              NaN   
3487         거제시장  상설장     매일  35.181932  129.072793              NaN   
3488         연동시장  상설장     매일  35.188843  129.092523              NaN   

     공중화장실보유여부 주차장보유여부  개설연도     데이터기준일자       제공기관명  
0            Y       N  1990  2020-02-25        경상북도  
1            Y       N  1986  2020-02-25        경상북도  
2            Y       Y  1981  2020-02-25        경상북도  
3            Y       Y  1982  2020-02-25        경상북도  
4            Y       N  1985  2020-02-25        경상북도  
...        ...     ...   ...         ...         ...  
3484         Y       Y   NaN  2021-10-26  소상공인시장진흥공단  
3485         Y       Y   NaN  2021-10-26  소상공인시장진흥공단  
3486         Y       N   NaN  2021-10-26  소상공인시장진흥공단  
3487         Y       N   NaN  2021-10-26  소상공인시장진흥공단  
3488         Y       Y   NaN  2021-10-26  소상공인시장진흥공단  

[3411 rows x 11 columns]

#### 03. 위치 정보를 이용한 시각화

In [12]:
market_name = market_2['시장명']
market_lati = market_2['위도']
market_long = market_2['경도']

In [13]:
import numpy as np

market_lati = list(market_lati)
market_long = list(market_long)
market_loc = np.array([market_lati, market_long]).T
print( market_loc.shape )
print( np.mean( market_lati) , np.mean(market_long)  )
market_loc

(3411, 2)
36.20206678435737 127.85249836869701


array([[ 36.0127023 , 129.366493  ],
       [ 36.01993305, 129.36110774],
       [ 36.02580836, 129.3709491 ],
       ...,
       [ 35.19000574, 129.0682199 ],
       [ 35.18193157, 129.0727929 ],
       [ 35.18884338, 129.0925228 ]])

In [14]:
from folium import plugins
import os

market_map = folium.Map(location=[ np.mean( market_lati), np.mean(market_long) ], 
                       zoom_start=8)

market_name = list(market_name)
plugins.MarkerCluster(market_loc, popups=market_name).add_to(market_map)

market_map.save(os.path.join('.', 'Market_location.html'))
market_map